enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,193 @@
+* The deployment request makes the same API call to Nova as with a virtual machine install (there are no explicit calls to Ironic or anything in the Installer required)
+  * so creating baremetal nodes should work transparently from the installer as well as Cluster API provider OpenStack
+* both virtual and baremetal machines consume the same image from glance
+
+### Probable issues
+
+* When attaching and detaching, the expected timings must be adjusted to
+  reflect the different boot and shutdown latencies of physical machines
+
+* When attaching, the Certificate Signing Request timeout must be adjusted to
+  reflect the different boot and shutdown latencies of physical machines
+
+### User Stories
+
+#### Increased performance
+
+User wants to run the Compute Nodes on baremetal machines for increased
+performance. The Control Plane sits on regular Nova instances to leverage the
+flexibility of virtual machines.
+
+### Risks and Mitigations
+
+**Uncertainty about OpenStack's extra networking resources**
+
+Are ports (including trunk), security groups, allowed address pairs all these necessary for baremetal?
+
+Will the Cluster-API-Provider be able to set them on baremetal-based nodes? Is it even possible to assign a port or a security group to a baremetal machine?
+
+A process where nodes are manually created and attached, such as what described in the UPI documentation, should provide a mitigation.
+
+## Design Details
+
+### Test Plan
+
+The enhancement will be tested using virtual baremetal machines.
+
+TODO: alternatives
+	- OVB on RDO?
+        - MOC is planning on adding Ironic
+	- Install (and snapshot?) an Ironic-enabled OSP on PSI (see Infrared-upshift)
+
+
+TODO: references
+        - https://gitlab.cee.redhat.com/jstransk/infrared-upshift-guide
+	- https://docs.openstack.org/project-deploy-guide/tripleo-docs/latest/environments/virtualbmc.html
+        - https://openstack-virtual-baremetal.readthedocs.io/en/latest/introduction.html
+
+### Graduation Criteria
+
+TBD
+
+##### Removing a deprecated feature
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+TBD
+
+### Version Skew Strategy
+
+TBD
+
+## Implementation History
+
+TBD
+
+## Drawbacks
+
+TBD
+
+## Alternatives
+
+Being able to run at least some workloads on baremetal nodes is highly
+desirable for certain users. It should be possible to achieve that
+even without an explicit support:
+
+### Baremetal IPI / UPI
+
+Support for an installer-provisioned infrastructure deployments where
+all the nodes are baremetal is currently being developed.
+Baremetal-based user-provisioned infrastructure deployments are
+supported today.
+
+Both approaches are all-or-nothing. Every workload running on the
+OpenShift cluster will run on a baremetal node. This may get wasteful
+and expensive.
+
+Integrating the baremetal approach with an OpenStack cloud can also be
+more complicated.
```

Comment:
This last statement feels like it should be expanded because I'm not sure what exactly is trying to be communicated.